jdbc-driver/src/main/java/com/google/cloud/dataproc/jdbc/HiveUrlUtils.java

```diff
@@ -0,0 +1,158 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.dataproc.jdbc;
+
+import com.google.common.collect.ImmutableSet;
+import java.io.IOException;
+import java.net.URI;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/** Helper methods to establish Connection for DataprocDriver. */
+public class HiveUrlUtils {
+    public static final String HIVE_DEFAULT_DATABASE = "";
+    public static final int HIVE_DEFAULT_PORT = 10001;
+    public static final String URL_JDBC_PREFIX = "jdbc:";
+
+    /**
+     * Parses client url and extracts Hive connection parameters.
+     *
+     * @param url verified url passed in by client, already of valid format possible formats:
+     *     jdbc:dataproc://hive/<dbName>;clusterName=<>;projectId=<>;region=<>;port=<>
+     *     jdbc:dataproc://hive/<dbName>;clusterName=<>;projectId=<>;region=<>;clusterPoolLabel=foo=bar:env=staging;
+     * @return HiveJdbcConnectionParam object with fields correctly set
+     * @throws IOException
+     * @throws InvalidURLException
+     */
+    public static HiveJdbcConnectionOptions parseHiveUrl(String url) throws InvalidURLException {
+        HiveJdbcConnectionOptions.Builder paramBuilder = HiveJdbcConnectionOptions.builder();
+
+        // Stripped out for the purpose of using URI
+        String urlStripped = url.substring(URL_JDBC_PREFIX.length());
+        URI jdbcURI = URI.create(urlStripped);
+
+        // getPath() will return: /<dbName>;<sessionConfs>, substring gets rid of the "/"
+        String dbAndSessionConfs = jdbcURI.getPath().substring(1);
+        // getQuery() will return the info after '?'
+        String hiveConfs = jdbcURI.getQuery();
+        // getFragmeng() will return the info after '#'
+        String hiveVars = jdbcURI.getFragment();
+
+        String[] params = dbAndSessionConfs.split(";");
+        // Incorrect database name provided
+        if (params.length != 0 && params[0].contains("=")) {
+            throw new InvalidURLException(
+                    String.format(
+                            "'%s' Database name must not contain '='. "
+                                    + "If no dbName is specified, format should be: %s",
+                            params[0], "jdbc:dataproc://hive/;clusterName=<clusterName>"));
+        }
+
+        paramBuilder.setDbName(params.length == 0 ? HIVE_DEFAULT_DATABASE : params[0]);
+        // Params that do not need parsing
+        paramBuilder.setHiveConfs(hiveConfs);
+        paramBuilder.setHiveVars(hiveVars);
+
+        Map<String, String> paramsMap = paramsToMap(params);
+
+        // Get otherSessionConfigs
+        Set<String> nonSessionConfs =
```

Comment:
Extract this map as a class-level constant and convert name to CAPITALIZED_SNAKE_CASE.